exo.kernel.component.common/src/main/java/org/exoplatform/services/scheduler/JobInfo.java

```diff
@@ -47,6 +49,19 @@ public JobInfo(String jobName, String groupName, Class job)
       groupName_ = groupName;
    }
 
+   public JobInfo(String jobName_, String groupName_, Class job_, String shouldRecover)
+   {
+      this(jobName_, groupName_, job_);
+      if(shouldRecover == null)
+      {
+         this.shouldRecover= true;
+      }
+      else
+      {
+         this.shouldRecover = Boolean.getBoolean(shouldRecover);
```

Comment:
you should use Boolean.valueOf instead of Boolean.getBoolean.
In fact this complete if section is not necessary with valueOf